### ISSUE TITLE
Multi node training of YOLOv8 Stuck in the following steps

### ISSUE THREAD

**Issue by Python123wbl456** [ultralytics/ultralytics#11764]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

work0
(ddp) (base) wbl@ubuntu01:~$ torchrun --nproc_per_node 1 --nnodes 2 --node_rank 0 --master_addr "172.17.141.98" --master_port 8547 --local_addr "172.17.141.98" /home/wbl/ultralytics-8.2.0/train.py
RANK=0 -- LOCAL_RANK=0
0 0
About to load model: /home/wbl/ultralytics-8.2.0/yolov8n.pt
About to train model: /home/wbl/ultralytics-8.2.0/yolov8n.pt
Ultralytics YOLOv8.2.0 🚀 Python-3.10.14 torch-2.2.2 CUDA:0 (NVIDIA GeForce RTX 3090, 24253MiB)
Overriding model.yaml nc=80 with nc=1
Transferred 319/355 items from pretrained weights
COMET WARNING: Can not parse empty Comet API key
WARNING ⚠️ Comet installed but not initialized correctly, not logging this run. Comet.ml requires an API key. Please provide as the first argument to Experiment(api_key) or as an environment variable named COMET_API_KEY 
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning /home/wbl/ultralytics-8.2.0/datasets/labels/new_train.cache... 900 images, 0 backgrounds, 0 corrupt: 100%|██████████| 900/900 [00:00<?, ?it/s]
ubuntu01:2366657:2366657 [0] NCCL INFO NCCL_SOCKET_IFNAME set by environment to eno4
ubuntu01:2366657:2366657 [0] NCCL INFO Bootstrap : Using eno4:172.17.141.98<0>
ubuntu01:2366657:2366657 [0] NCCL INFO NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation
ubuntu01:2366657:2366657 [0] NCCL INFO cudaDriverVersion 12040
NCCL version 2.19.3+cuda12.1
ubuntu01:2366657:2366696 [0] NCCL INFO Failed to open libibverbs.so[.1]
ubuntu01:2366657:2366696 [0] NCCL INFO NCCL_SOCKET_IFNAME set by environment to eno4
ubuntu01:2366657:2366696 [0] NCCL INFO NET/Socket : Using [0]eno4:172.17.141.98<0>
ubuntu01:2366657:2366696 [0] NCCL INFO Using non-device net plugin version 0
ubuntu01:2366657:2366696 [0] NCCL INFO Using network Socket
ubuntu01:2366657:2366696 [0] NCCL INFO comm 0x241441b0 rank 0 nranks 2 cudaDev 0 nvmlDev 0 busId 65000 commId 0x7fe2e5a0b97c69b9 - Init START
ubuntu01:2366657:2366696 [0] NCCL INFO Setting affinity for GPU 0 to ffff
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 00/02 :    0   1
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 01/02 :    0   1
ubuntu01:2366657:2366696 [0] NCCL INFO Trees [0] 1/-1/-1->0->-1 [1] -1/-1/-1->0->1
ubuntu01:2366657:2366696 [0] NCCL INFO P2P Chunksize set to 131072
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 00/0 : 1[0] -> 0[0] [receive] via NET/Socket/0
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 01/0 : 1[0] -> 0[0] [receive] via NET/Socket/0
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 00/0 : 0[0] -> 1[0] [send] via NET/Socket/0
ubuntu01:2366657:2366696 [0] NCCL INFO Channel 01/0 : 0[0] -> 1[0] [send] via NET/Socket/0
ubuntu01:2366657:2366696 [0] NCCL INFO Connected all rings
ubuntu01:2366657:2366696 [0] NCCL INFO Connected all trees
ubuntu01:2366657:2366696 [0] NCCL INFO threadThresholds 8/8/64 | 16/8/64 | 512 | 512
ubuntu01:2366657:2366696 [0] NCCL INFO 2 coll channels, 0 nvls channels, 2 p2p channels, 2 p2p channels per peer
ubuntu01:2366657:2366696 [0] NCCL INFO comm 0x241441b0 rank 0 nranks 2 cudaDev 0 nvmlDev 0 busId 65000 commId 0x7fe2e5a0b97c69b9 - Init COMPLETE
val: Scanning /home/wbl/ultralytics-8.2.0/datasets/labels/new_val.cache... 100 images, 0 backgrounds, 0 corrupt: 100%|██████████| 100/100 [00:00<?, ?it/s]
work1
(ddp) (base) wbl@ubuntu02:~$ torchrun --nproc_per_node 1 --nnodes 2 --node_rank 1 --master_addr "172.17.141.98" --master_port 8547 --local_addr "172.17.141.99" /home/wbl/ultralytics-8.2.0/train.py
RANK=1 -- LOCAL_RANK=0
1 0
About to load model: /home/wbl/ultralytics-8.2.0/yolov8n.pt
About to train model: /home/wbl/ultralytics-8.2.0/yolov8n.pt
train: Scanning /home/wbl/ultralytics-8.2.0/datasets/labels/new_train.cache... 900 images, 0 backgrounds, 0 corrupt: 100%|██████████| 900/900 [00:00<?, ?
ubuntu02:2886370:2886370 [0] NCCL INFO cudaDriverVersion 12040
ubuntu02:2886370:2886370 [0] NCCL INFO NCCL_SOCKET_IFNAME set by environment to eno4
ubuntu02:2886370:2886370 [0] NCCL INFO Bootstrap : Using eno4:172.17.141.99<0>
ubuntu02:2886370:2886370 [0] NCCL INFO NET/Plugin : dlerror=libnccl-net.so: cannot open shared object file: No such file or directory No plugin found (libnccl-net.so), using internal implementation
ubuntu02:2886370:2886409 [0] NCCL INFO Failed to open libibverbs.so[.1]
ubuntu02:2886370:2886409 [0] NCCL INFO NCCL_SOCKET_IFNAME set by environment to eno4
ubuntu02:2886370:2886409 [0] NCCL INFO NET/Socket : Using [0]eno4:172.17.141.99<0>
ubuntu02:2886370:2886409 [0] NCCL INFO Using non-device net plugin version 0
ubuntu02:2886370:2886409 [0] NCCL INFO Using network Socket
ubuntu02:2886370:2886409 [0] NCCL INFO comm 0x70b5d40 rank 1 nranks 2 cudaDev 0 nvmlDev 0 busId 65000 commId 0x7fe2e5a0b97c69b9 - Init START
ubuntu02:2886370:2886409 [0] NCCL INFO Setting affinity for GPU 0 to ffff
ubuntu02:2886370:2886409 [0] NCCL INFO Trees [0] -1/-1/-1->1->0 [1] 0/-1/-1->1->-1
ubuntu02:2886370:2886409 [0] NCCL INFO P2P Chunksize set to 131072
ubuntu02:2886370:2886409 [0] NCCL INFO Channel 00/0 : 0[0] -> 1[0] [receive] via NET/Socket/0
ubuntu02:2886370:2886409 [0] NCCL INFO Channel 01/0 : 0[0] -> 1[0] [receive] via NET/Socket/0
ubuntu02:2886370:2886409 [0] NCCL INFO Channel 00/0 : 1[0] -> 0[0] [send] via NET/Socket/0
ubuntu02:2886370:2886409 [0] NCCL INFO Channel 01/0 : 1[0] -> 0[0] [send] via NET/Socket/0
ubuntu02:2886370:2886409 [0] NCCL INFO Connected all rings
ubuntu02:2886370:2886409 [0] NCCL INFO Connected all trees
ubuntu02:2886370:2886409 [0] NCCL INFO threadThresholds 8/8/64 | 16/8/64 | 512 | 512
ubuntu02:2886370:2886409 [0] NCCL INFO 2 coll channels, 0 nvls channels, 2 p2p channels, 2 p2p channels per peer
ubuntu02:2886370:2886409 [0] NCCL INFO comm 0x70b5d40 rank 1 nranks 2 cudaDev 0 nvmlDev 0 busId 65000 commId 0x7fe2e5a0b97c69b9 - Init COMPLETE

**Comment by glenn-jocher** [comment]
@Python123wbl456 it looks like your multi-node training setup is mostly correct, but the logs indicate a potential issue with the NCCL communication library, specifically with the failure to open `libnccl-net.so` and `libibverbs.so`. These libraries are essential for efficient GPU communication in multi-node setups.

Here are a couple of steps you can take to resolve this:

1. **Ensure NCCL is installed correctly**: Verify that NCCL is installed and correctly configured on both nodes. You might need to reinstall or update NCCL.

2. **Check Network Configuration**: Ensure that the network interface specified (`NCCL_SOCKET_IFNAME=eno4`) is correct and consistent across all nodes. You can check available interfaces using `ifconfig`.

3. **Permissions and Paths**: Check that the paths to NCCL libraries are correctly set and that your user has the necessary permissions to access them.

If these steps don't resolve the issue, you might want to consider posting a more detailed question with specific error logs on platforms like Stack Overflow or NVIDIA's developer forums, where a community of developers can help troubleshoot NCCL-specific issues.
